src/common/components/nav/nav.component.js

```diff
@@ -0,0 +1,70 @@
+import angular from 'angular';
+import {Observable} from 'rxjs/Observable';
+import 'rxjs/add/operator/map';
+
+import cartService from 'common/services/api/cart.service';
+import loadingComponent from 'common/components/loading/loading.component';
+import signInButton from 'common/components/signInButton/signInButton.component';
+
+import mobileTemplate from './mobileNav.tpl';
+
+let componentName = 'nav';
+
+class NavController{
+
+  /* @ngInject */
+  constructor($http, $document, envService, cartService){
+    this.$http = $http;
+    this.$document = $document;
+    this.cartService = cartService;
+    this.imgDomain = envService.read('imgDomain');
+    this.navFeed = envService.read('navFeed');
+  }
+
+  $onInit() {
+    this.getNav().subscribe((structure) => {
+      this.menuStructure = structure;
+    });
+  }
+
+  getNav() {
+    return Observable.from(this.$http({
+      method: 'GET',
+      url: this.navFeed,
+      headers: {
+        'Accept': undefined
```

Comment:
can you pass 'application/json' here to have $http automatically parse the JSON?